pkg/serverless/executioncontext/executioncontext.go

```diff
@@ -56,17 +61,34 @@ func (ec *ExecutionContext) GetCurrentState() State {
 		ARN:                ec.arn,
 		LastRequestID:      ec.lastRequestID,
 		ColdstartRequestID: ec.coldstartRequestID,
+		WasColdStart:       ec.wasColdStart,
+		WasProactiveInit:   ec.wasProactiveInit,
 		LastLogRequestID:   ec.lastLogRequestID,
 		LastOOMRequestID:   ec.lastOOMRequestID,
 		Runtime:            ec.runtime,
-		Coldstart:          ec.coldstart,
-		ProactiveInit:      ec.proactiveInit,
 		InitTime:           ec.initTime,
 		StartTime:          ec.startTime,
 		EndTime:            ec.endTime,
 	}
 }
 
+// Returns whether or not the given request ID is a cold start or is a proactive init
+func (ec *ExecutionContext) GetColdStartTagsForRequestID(requestID string) ColdStartTags {
+	ec.m.Lock()
+	defer ec.m.Unlock()
+	coldStartTags := ColdStartTags{
+		IsColdStart:     false,
+		IsProactiveInit: false,
+	}
+	if requestID != ec.coldstartRequestID {
+		return coldStartTags
```

Comment:
I guess this is inferring since they are not the same request, then the current one must not be a cold start?

Is it possible for an invocation to not finish (which might need to be flushed on the next one), but the subsequent invocation to trigger a cold start?

Reply:
It could if it failed to initialize, but I think the request ID isn't even set then so it wouldn't matter.